### PR TITLE
Plumb failure accrual into the grpc api

### PR DESCRIPTION
**Pull request by adleong**
Updates #10655 to plumb failure accrual configuration through from annotations into the outbound policy proxy API.

Tested manually by setting various combinations of these annotations on Services:
```
    balancer.linkerd.io/failure-accrual-consecutive-failures
    balancer.linkerd.io/max-penalty
    balancer.linkerd.io/min-penalty
    balancer.linkerd.io/mode
```

and watching the output from a manual query of the outbound policy API:

```
grpcurl --plaintext --proto outbound.proto -d '{"source_workload": "emojivoto:emoji-5f947b5dbd-b6rkl", "addr": {"port": 7002, "ip": {"ipv4": 174090160}}}' localhost:8090 io.linkerd.proxy.outbound.OutboundPolicies.Watch
```

Next steps would be to add some tests for this in the outbound API tests.

### CODE REVIEW

File: policy-controller/k8s/index/src/outbound/index.rs [hunk 1]
```diff
@@ -435,71 +440,65 @@ impl ServiceRoutes {
...
-                    s.parse::<f64>()
-                        .map_err(|error| tracing::debug!(%error, "Invalid annotation value"))
-                        .ok()
-                })
-                .unwrap_or_default();
-            if min_penalty > max_penalty {
-                tracing::debug!(?min_penalty, ?max_penalty, "Invalid backoff configuration");
-                return Some(FailureAccrual::Consecutive {
+    annotations
+        .get("balancer.linkerd.io/mode")
```
Comment (olix0r) [inline policy-controller/k8s/index/src/outbound/index.rs:456]: I don't think this annotation name is appropriate. We should clearly scope this to failure accrual as `failure-accrual-mode`. `balancer.linkerd.io/mode=consecutive` strongly suggests that we're using some sort of "consecutive" balancing strategy.

**Review by olix0r — CHANGES_REQUESTED** [review]

File: policy-controller/k8s/index/src/outbound/index.rs (hunk 1 above)
Comment (mateiidavid) [reply policy-controller/k8s/index/src/outbound/index.rs:456]: That's on me, these were just placeholders I used when I started working on it Friday morning. The intent was to change them.

File: policy-controller/k8s/index/src/outbound/index.rs [hunk 2]
```diff
@@ -435,71 +440,65 @@ impl ServiceRoutes {
...
-                    min_penalty,
-                    max_penalty,
-                    jitter,
-                },
-            })
-        }
-        _ => None,
-    }
+        })
+        .transpose()
```
Comment (mateiidavid) [inline policy-controller/k8s/index/src/outbound/index.rs:501]: nice, combinator approach looks cleaner than the match I had come up with 👌🏻

**Review by mateiidavid — APPROVED** [review]

File: policy-controller/k8s/index/src/outbound/index.rs [hunk 3]
```diff
@@ -435,71 +440,65 @@ impl ServiceRoutes {
...
+            if mode == "consecutive" {
+                let max_failures = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-failures")
+                    .map(|s| s.parse::<u32>())
+                    .transpose()?
+                    .unwrap_or(7);
+
+                let max_penalty = annotations
+                    .get("balancer.linkerd.io/max-penalty")
+                    .map(|s| parse_duration(s))
```
Comment (hawkw) [inline policy-controller/k8s/index/src/outbound/index.rs:467]: style nit, TIOLI: could just be
```suggestion
                    .map(parse_duration)
```

File: policy-controller/k8s/index/src/outbound/index.rs [hunk 4]
```diff
@@ -435,71 +440,65 @@ impl ServiceRoutes {
...
+                    .get("balancer.linkerd.io/jitter")
+                    .map(|s| s.parse::<f64>())
+                    .transpose()?
+                    .unwrap_or_default();
+                if min_penalty > max_penalty {
+                    bail!(
+                        "min_penalty ({:?}) cannot exceed max_penalty ({:?})",
+                        min_penalty,
+                        max_penalty,
+                    );
```
Comment (hawkw) [inline policy-controller/k8s/index/src/outbound/index.rs:486]: style nit, take it or leave it: could be
```suggestion
                    bail!("min_penalty ({min_penalty:?}) cannot exceed max_penalty ({max_penalty:?})");
```
(assuming rustfmt doesn't want to wrap it)

File: policy-controller/k8s/index/src/outbound/index.rs [hunk 5]
```diff
@@ -435,71 +440,65 @@ impl ServiceRoutes {
...
-                    backoff: Default::default(),
-                });
+                    backoff: Backoff {
+                        min_penalty,
+                        max_penalty,
+                        jitter,
+                    },
+                })
+            } else {
+                bail!("unsupported failure accrual mode: {}", mode);
```
Comment (hawkw) [inline policy-controller/k8s/index/src/outbound/index.rs:498]: TIOLI: could be
```suggestion
                bail!("unsupported failure accrual mode: {mode}");
```

**Review by hawkw — APPROVED** [review]
left some driveby style nits, none of which are actually important

**Pull request merged**